frontend/lib/resolution/copy-elision.cpp

```diff
@@ -221,12 +226,9 @@ void FindElidedCopies::addCopyInit(VarFrame* frame, ID fromVarId, ID point) {
   state.points.insert(point);
 }
 void FindElidedCopies::addMention(VarFrame* frame, ID varId) {
-  auto it = frame->copyElisionState.find(varId);
-  if (it != frame->copyElisionState.end()) {
-    CopyElisionState& state = it->second;
-    state.lastIsCopy = false;
-    state.points.clear();
-  }
+  CopyElisionState& state = frame->copyElisionState[varId];
+  state.lastIsCopy = false;
+  state.points.clear();
```

Comment:
I'm still curious about this. The reason the code originally didn't populate the map if it didn't already have an entry, is because (no ID in copyElisionState) => (the ID has no elision points and the last mention is not a copy). Thus, the code setting stuff on lines 229-231 is almost the same as not having an entry at all. I assume this is a memory use optimization.

Is there a reason this stopped working? What might it be -- it's not obvious to me on my read-through.

Reply:
iirc the problem was with mentions of variables getting propagated to outer scopes. So, currently, this allows copy elision:
```
proc foo() {
var x = 0;
var y = x;
{x;}
}
```
because the mention of x in the inner scope wasn't making it back up to the parent frame.